tfe/data_source_organization_membership.go

```diff
@@ -40,29 +40,18 @@ func dataSourceTFEOrganizationMembershipRead(d *schema.ResourceData, meta interf
 	// Create an options struct.
 	options := &tfe.OrganizationMembershipListOptions{
 		Include: []tfe.OrgMembershipIncludeOpt{tfe.OrgMembershipUser},
+		Emails:  []string{email},
 	}
 
-	for {
-		organizationMembershipsList, err := tfeClient.OrganizationMemberships.List(ctx, organization, options)
-		if err != nil {
-			return fmt.Errorf("Error retrieving organization memberships: %v", err)
-		}
-
-		for _, organizationMembership := range organizationMembershipsList.Items {
-			if organizationMembership.User.Email == email {
-				d.SetId(organizationMembership.ID)
-				return resourceTFEOrganizationMembershipRead(d, meta)
-			}
-		}
-
-		// Exit the loop when we've seen all pages.
-		if organizationMembershipsList.CurrentPage >= organizationMembershipsList.TotalPages {
-			break
-		}
+	oml, err := tfeClient.OrganizationMemberships.List(ctx, organization, options)
+	if err != nil {
+		return fmt.Errorf("Error retrieving organization memberships: %v", err)
+	}
 
-		// Update the page number to get the next page.
-		options.PageNumber = organizationMembershipsList.NextPage
+	if len(oml.Items) == 0 {
+		return fmt.Errorf("Could not find organization membership for organization %s and email %s", organization, email)
 	}
 
-	return fmt.Errorf("Could not find organization membership for organization %s and email %s", organization, email)
+	d.SetId(oml.Items[0].ID)
```

Comment:
What if this is used against a TFE version before this filter did anything? I think the brute search needs to be a fallback?